### PR TITLE
Fix Format syntax err message when value out of range

### PR DESCRIPTION
**Pull request by chenquan**
### What problem does this PR solve? 
related PR: pingcap/tidb#17489 pingcap/tidb#17745

### What is changed and how it works?
Ignore error that the decimal  out  of range the value 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by chenquan** [comment]
Why are all integration tests timed out?
https://app.circleci.com/pipelines/github/pingcap/parser
image

**Comment by chenquan** [comment]
PTAL @tangenta @kennytm @JmPotato

File: yy_parser.go [hunk 1]
```diff
@@ -158,7 +158,24 @@ func (parser *Parser) Parse(sql, charset, collation string) (stmt []ast.StmtNode
 		warns = nil
 	}
 	if len(errs) != 0 {
-		return nil, warns, errors.Trace(errs[0])
+		err := errs[0]
```
Comment (xiongjiwei) [inline yy_parser.go:161]: It changes the logistic, can you please explain why do not return the error in some situations?

**Comment by xiongjiwei** [comment]
BTW, can you please add some tests for it?

File: yy_parser.go (hunk 1 above)
Comment (chenquan) [reply yy_parser.go:161]: Maybe it's not the best solution, I'll close it.

**Pull request closed without merging**